source/connection.c

```diff
@@ -617,8 +629,30 @@ static void s_server_bootstrap_on_accept_channel_shutdown(
 static void s_server_bootstrap_on_server_listener_destroy(struct aws_server_bootstrap *bootstrap, void *user_data) {
     (void)bootstrap;
     AWS_ASSERT(user_data);
-    struct aws_http_server *server = user_data;
+    struct aws_server_user_data *server_user_data = user_data;
+    struct aws_http_server *server = server_user_data->server;
     s_http_server_clean_up(server);
+    aws_mem_release(server_user_data->alloc, server_user_data);
```

Comment:
You're probably leaking `server_user_data` if this fails in the async-setup phase.

Also, forgot to call aws_mutex_clean_up() and aws_condition_variable_clean_up(), but now need to call aws_future_void_destroy(). Probably want to do this cleanup in a helper function ... oh hey there's already an s_http_server_clean_up() function. Probably should alter it to take `aws_server_user_data  *` instead of `aws_http_server *`, since the server always lives inside the user-data struct now